### ISSUE TITLE
IDE type issues with $device helper

### ISSUE THREAD

**Issue by gregmulvaney** [nuxt-modules/device#136]
Webstorm is unable to locate the types for the $device helper. Not really familiar with Nuxt plugins/modules so unfortunately I'm not able to locate the issue. App fails nuxi typecheck as well but is able to be built and functions as expected. Tried adding the module to my tsconfig with no fix. 

Nuxt: v3.2.2
device: v3.1.0

**Comment by danielroe** [comment]
~> https://github.com/nuxt/nuxt/issues/19399

**Comment by gregmulvaney** [comment]
Unless Webstorm is changing something under the hood, I can't see why I would also fail `nuxi typecheck` with the error of 

`error TS2339: Property '$device' does not exist on type '{ $: ComponentInternalInstance; $data: {}; $props: Partial<{}> & Omit<Readonly<ExtractPropTypes<{}>> & VNodeProps & AllowedComponentProps & ComponentCustomProps, never>; ... 10 more ...; $watch<T extends string | ((...args: any) => any)>(source: T, cb: T extends (...args: any) => infer R ? (args_0: R, args_1: R) => ...'.`

**Comment by danielroe** [comment]
No, that shouldn't happen. Would you provide a reproduction?

**Comment by gregmulvaney** [comment]
Sure can!

Here is the sandbox: https://codesandbox.io/p/sandbox/great-bash-7zdlg1

Issue persist when running `pnpx nuxi typecheck`

**Comment by danielroe** [comment]
That's a bug with this module, as it does not generate types for its plugin: https://unpkg.com/browse/@nuxtjs/device@3.1.0/dist/runtime/plugin.d.ts.

**Comment by nathanchase** [comment]
Yeah, I get a similar error in VSCode:
`'__VLS_ctx.$device' is of type 'unknown'.`

**Comment by joseehilton147** [comment]
yep, same error:

![image](https://user-images.githubusercontent.com/53986027/226642665-22a061f7-147f-4d53-9c1e-7136227500a0.png)

**Comment by max-arias** [comment]
I came across the same issue, I shimmed it by doing:

```
import { ComponentCustomProperties } from 'vue';

type Device = {
  userAgent: string;
  isDesktop: boolean;
  isIos: boolean;
  isAndroid: boolean;
  isMobile: boolean;
  isMobileOrTablet: boolean;
  isDesktopOrTablet: boolean;
  isTablet: boolean;
  isWindows: boolean;
  isMacOS: boolean;
  isApple: boolean;
  isSafari: boolean;
  isFirefox: boolean;
  isEdge: boolean;
  isChrome: boolean;
  isSamsung: boolean;
  isCrawler: boolean;
};

declare module '@vue/runtime-core' {
  interface ComponentCustomProperties {
    $device: Device;
  }
}
```

**Comment by popdo** [comment]
I get a similar error in VSCode:

'__VLS_ctx.$device' is of type 'unknown'.

**Comment by sholohov** [comment]
I solved the problem for myself like this
```ts
// plugins.d.ts

import type { Device } from "@nuxtjs/device/runtime/types";

declare module '@vue/runtime-core' {
  interface ComponentCustomProperties {
    $device: Device;
  }
}
```

**Comment by steklopod** [comment]
> I solved the problem for myself like this
> 
> ```ts
> // plugins.d.ts
> 
> import type { Device } from "@nuxtjs/device/runtime/types";
> 
> declare module '@vue/runtime-core' {
>   interface ComponentCustomProperties {
>     $device: Device;
>   }
> }
> ```

That helped, but would be nice to have it out of the box in library

**Comment by DamianGlowala** [comment]
This should be now fixed in v3.2.2.
